### DISCUSSION TITLE
Is it possible to access context inside a computed field?

### DISCUSSION THREAD

**Discussion by devjourney24** [pydantic/pydantic#6415] [Pydantic V2]
Hi,
The Pydantic v2 is awesome, thank you for the great work! 🚀

Starting a new project with it.
Is there any way to access context (passed in model_validate) inside a computed field? 

Or maybe there is some better way to achieve the goal?

```python

class ItemImage(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    image_path: str

    @computed_field
    @property
    def full_image_url(self) -> str:
        return f'{s3_bucket_url_from_somewhere}{self.image_path}'

class Item(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    item_images: list[ItemImage] 
```

I'm converting the SQLAlchemy models (Item with ItemImage relationship) and want to pass the base S3 bucket url on that step:

```python
result = await self.db_session.scalars(select(Item).filter(Item.id == id).options(selectinload(Item.item_images)))
item = result.one()
return Item.model_validate(item, context={'s3_bucket_url': settings.s3_base_url})
```

I know I can solve it by using global settings var directly inside computed_field, but I'm trying to avoid it to keep everything clean.

**Comment by BredoGen** [comment]
You can use a workaround with a field validator where context is available. But I agree computed field will be better.

```python
class ItemImage(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    image_path: str
    full_image_url: str = Field(default='', validate_default=True)

    @field_validator('full_image_url', mode='before')
    def generate_image_url(cls, v, info: FieldValidationInfo):
        context = info.context
        return f"{context.get('s3_bucket_url')}/{info.data['image_path']}"
```

**Reply by devjourney24** [reply]
Thanks, but this looks like very v1 style, where we were using validators as computed field.
Now when computed fields are available in v2, it'll be great to have context there too.

**Comment by andrew-bsp** [comment]
Would this other workaround approach be sensible, to make that piece of context a proper field of the model, with a default value that comes from context? Then it can be used from any computed fields that want it:

```python
class ItemImage(BaseModel):
    id: int
    image_path: str
    s3_bucket_url: str = Field(default=None, validate_default=True, exclude=True)

    @field_validator("s3_bucket_url", mode="before")
    def get_s3_bucket_from_context(cls, v, info):
        assert v is None
        return info.context.get("s3_bucket_url")

    @computed_field
    def full_image_url(self) -> str:
        return f'{self.s3_bucket_url}{self.image_path}'
```
